README.md

```diff
@@ -1102,3 +1102,26 @@ To enable fast minify mode with the API use:
 ```js
 UglifyJS.minify(code, { compress: false, mangle: true });
 ```
+
+#### Source maps and debugging
+
+Various `compress` transforms that simplify, rearrange, inline and remove code
+are known to have an adverse effect on debugging with source maps. This is
+expected as code is optimized and mappings are often simply not possible as
+some code no longer exists. For highest fidelity in source map debugging
+disable the Uglify `compress` option and just use `mangle`.
+
+### Compiler assumptions
+
+To allow for better optimizations, the compiler makes various assumptions.
+
+- `.toString()` and `.valueOf()` have not been overridden, and don't have side effects.
```

Comment:
`toString()` *can* be overridden in new functions/classes - it just cannot have side effects.

Reply:
Do you have a suggestion for how to phrase this? My take:

> `.toString()` and `.valueOf()` don't have side effects, and for built-in objects they have not been overridden.

Reply:
... only if `unsafe` is not specified, though I guess that's covered under https://github.com/mishoo/UglifyJS2#the-unsafe-compress-option

Reply:
https://github.com/mishoo/UglifyJS2/pull/2883/commits/a0d408ea3d7fca7333d625ecbec27c4e0e3903d1 is good.